content/en/blog/_posts/2020-05-21-wsl2-dockerdesktop-k8s.md

```diff
@@ -0,0 +1,584 @@
+
+![image-20200323223208416](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-kind-browse-dashboard-browse-nodes.png)
+
+A nice and shiny three nodes appear.
+
+# Minikube: Kubernetes from everywhere
+
+Right now, we have Docker that is installed, configured and the last test worked fine.
+
+However, if we look carefully at the `kubectl` command, it found the "Client Version" (1.15.5), but it didn't find any server.
+
+This is normal as we didn't enable the Docker Kubernetes cluster. So let's install Minikube and create our first cluster.
+
+And as sources are always important to mention, we will follow (partially) the how-to from the [Kubernetes.io website](https://kubernetes.io/docs/tasks/tools/install-minikube/):
+
+```bash
+# Download the latest version of Minikube
+curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
+# Make the binary executable
+chmod +x ./minikube
+# Move the binary to your executable path
+sudo mv ./minikube /usr/local/bin/
+```
+
+![image-20200328170504492](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-minikube-install.png)
+
+## Minikube: updating the host
+
+If we follow the how-to, it states that we should use the `--driver=none` flag in order to run Minikube directly on the host and Docker.
+
+Unfortunately, we will get an error about "conntrack" being required to run Kubernetes v 1.18:
+
+```bash
+# Create a minikube one node cluster
+minikube start --driver=none
+```
+
+![image-20200328171526437](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-minikube-start-error.png)
+
+> Tip: as you can see, the Terminal was changed so the nice icons are all displayed
+
+So let's fix the issue by installing the missing package:
+
+```bash
+# Install the conntrack package
+sudo apt install -y conntrack
+```
+
+![image-20200328174058258](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-minikube-install conntrack.png)
+
+Let's try to launch it again:
+
+```bash
+# Create a minikube one node cluster
+minikube start --driver=none
+# We got a permissions error > try again with sudo
+sudo minikube start --driver=none
+```
+
+![image-20200328202611584](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-minikube-start-error-systemd.png)
+
+Ok, this error cloud be problematic ... in the past. Luckily for us, there's a solution
+
+## Minikube: enabling SystemD
+
+In order to enable SystemD on WSL2, we will apply the [scripts](https://forum.snapcraft.io/t/running-snaps-on-wsl2-insiders-only-for-now/13033) from [Daniel Llewellyn](https://twitter.com/diddledan).
+
+I invite you to read the full blog post and how he came to the solution, and the various iterations he did to fix several issues.
+
+So in a nutshell, here are the commands:
+
+```bash
+# Install the needed packages
+sudo apt install -yqq daemonize dbus-user-session fontconfig
+```
+
+![image-20200328205328598](/images/blog/2020-05-21-wsl2-dockerdesktop-k8s/wsl2-minikube-systemd-packages.png)
+
+```bash

```

Comment:
This isn't quite a bash script. I think I'd separate the command to run `vi` from what the reader should paste in.
(also, personal taste, I'd use `sudoedit`).

Reply:
cc @nunix

Reply:
oh ok, by scripts I mean the `/usr/sbin/enter-systemd-namespace` and `/usr/sbin/start-systemd-namespace`.
for the first part, it's indeed not a script 👍 

for the separation, while I agree, I still have to see something that "appeals" or make sense to me. By typing the first command, the reader will enter in `vi`, so not really sure how to make a separation here.

As for `visudo`, sorry my old Unix habits remained here 😄